### DISCUSSION TITLE
How to Install Ansible Language server

### DISCUSSION THREAD

**Discussion by nippyin** [ansible/ansible-language-server#166] [Q&A] [answered]
Can we have a steps to install/configure it. I think it will help community at large. I'm trying to setup this on Nvim however unable to do with current instructions.

**Comment by webknjaz** [comment, marked as answer]
You should be able to learn this at https://github.com/yaegassy/coc-ansible.

Plus @yaegassy could probably answer this question better than me.

**Comment by mnarrell** [comment]
And if we're not using `coc`?

**Comment by muzammil-iftikhar** [comment]
Has anyone got it working. I am still stuck figuring it out since last night. The instructions at https://github.com/yaegassy/coc-ansible doesn't seem to be enough to get it working.
@yaegassy  @ssbarnea  can you support us here.

**Reply by yaegassy** [reply]
@muzammil-iftikhar Do you have python3, Node.js, coc.nvim installed in your environment? Have you installed and configured any plug-ins related to filetype? https://github.com/yaegassy/coc-ansible#-very-important--filetype-related

**Reply by muzammil-iftikhar** [reply]
@yaegassy,
coc.vim and filetype plugin
![image](https://user-images.githubusercontent.com/33807314/193217574-1a4f3025-7428-48d3-91c4-ce8d4a93d8fb.png)

python3 and nodejs
![image](https://user-images.githubusercontent.com/33807314/193217769-47a22e26-a027-4cf0-a09f-553d39c439ab.png)

Also, i did add following in my .vimrc file
`let g:coc_filetype_map = {
  \ 'yaml.ansible': 'ansible',
  \ }`

**Reply by yaegassy** [reply]
Try running `:echo &ft` when you open the YAML file for ansible. Does it correctly show `yaml.ansible`? Also, is coc.nvim itself working properly? If the Node.js version is very old, coc.nvim itself may not work, etc. Are extensions other than coc-ansible working?

**Reply by muzammil-iftikhar** [reply]
it shows yaml and i also see this error
![image](https://user-images.githubusercontent.com/33807314/193241984-413e665d-2dcf-4230-8e8c-6bc02c5f5ac6.png)

yes coc seems to be working fine for all these other extensions:
![image](https://user-images.githubusercontent.com/33807314/193242205-df7ca658-1db1-4d12-aaaa-77d8f2ce0609.png)

**Reply by yaegassy** [reply]
It looks like `coc-yaml` (use `yaml-language-server`) is running because the filetype is `yaml`, not `yaml.ansible`.

Please check the README and other documentation for "ansible-vim" and try adding the necessary settings. e.g. `g:ansible_ftdetect_filename_regex` https://github.com/pearofducks/ansible-vim

**Comment by muzammil-iftikhar** [comment]
Thanks @yaegassy. Working fine now
Just for others:
I used this to set the filetype to yaml.ansible
`au BufRead,BufNewFile */ansible/*.yml set filetype=yaml.ansible`
